signer/tests/integration/request_decider.rs

```diff
@@ -538,3 +538,182 @@ async fn blocklist_client_retry(num_failures: u8, failing_iters: u8) {
+
+    // We shouldn't have any decisions or votes at the beginning
+    if is_withdrawal {
+        let request_id = withdrawal_requests.first().unwrap().request_id;
+        let block_hash = &withdrawal_requests.first().unwrap().block_hash;
+        let qualified_id = withdrawal_requests.first().unwrap().qualified_id();
+        let decisions = db
+            .get_withdrawal_signers(request_id, block_hash)
+            .await
+            .unwrap();
+        let votes = db
+            .get_withdrawal_request_signer_votes(&qualified_id, &signer_public_key)
+            .await
+            .unwrap();
+        assert!(decisions.is_empty());
+        assert!(votes.iter().all(|vote| vote.is_accepted.is_none()));
+    } else {
+        let outpoint = deposit_requests.first().unwrap().outpoint();
+        let votes = db
+            .get_deposit_signers(&outpoint.txid.into(), outpoint.vout)
+            .await
+            .expect("Failed to read deposit signer votes");
+        let decisions = db
+            .get_deposit_signer_decisions(&chain_tip, 20, &signer_public_key)
+            .await
+            .expect("Failed to read deposit signer decisions");
+        assert!(votes.is_empty());
+        assert!(decisions.is_empty());
+    }
+
+    // Handle requests
+    request_decider.handle_new_requests().await.unwrap();
+
+    // Check that after requests handled we have votes and decisions, and that they are
+    // following blocklist
+    if is_withdrawal {
+        let request_id = withdrawal_requests.first().unwrap().request_id;
+        let block_hash = &withdrawal_requests.first().unwrap().block_hash;
+        let qualified_id = withdrawal_requests.first().unwrap().qualified_id();
+
+        let decisions = db
+            .get_withdrawal_signers(request_id, block_hash)
+            .await
+            .unwrap();
+        let votes = db
+            .get_withdrawal_request_signer_votes(&qualified_id, &signer_public_key)
+            .await
+            .unwrap();
+
+        assert_eq!(decisions.len(), 1);
+        assert_eq!(decisions.last().unwrap().is_accepted, !is_blocked);
+        assert!(!votes.is_empty());
+        assert!(votes.iter().any(|vote| {
+            vote.signer_public_key == signer_public_key && vote.is_accepted.unwrap() == !is_blocked
+        }));
+    } else {
+        let outpoint = deposit_requests.first().unwrap().outpoint();
+        let votes = db
+            .get_deposit_signers(&outpoint.txid.into(), outpoint.vout)
+            .await
+            .expect("Failed to read deposit signer votes");
+        let decisions = db
+            .get_deposit_signer_decisions(&chain_tip, 20, &signer_public_key)
+            .await
+            .expect("Failed to read deposit signer decisions");
+
+        assert_eq!(votes.len(), 1);
+        assert_eq!(votes.last().unwrap().can_accept, !is_blocked);
+        assert_eq!(decisions.len(), 1);
+        assert_eq!(decisions.last().unwrap().can_accept, !is_blocked);
```

Comment:
nit: using `is_blocked` as the variable is fine, but it feels more natural to use `is_accepted` as the variable and not have to use `!is_blocked` everywhere.